### ISSUE TITLE
Review handling of SAM 1.4+ CIGAR ops

### ISSUE THREAD

**Issue by bede** [bede/kindel#3]
Review handling of `N`, `H`, `P`, `X` and `=` ops. `X` and `=` were introduced in 1.4 and remove the need to concurrently examine the `MD` tag
https://samtools.github.io/hts-specs/SAMv1.pdf

**Comment by mdshw5** [comment]
Is this something that should be reviewed in mdshw5/simplesam as well?

**Comment by bede** [comment]
Hi @mdshw5  : )
Does Simplesam have operation-specific functionality? Or does it simply parse CIGAR ops as single char strings at face value? If the latter, it should be fine. BBMap is the only popular aligner implementing the 1.4 spec at this time AFAIK. So long as it handles `X` and `=` ops it'll be fine.

**Comment by mdshw5** [comment]
No, simplesam only uses the CIGAR operations to inspect the alignment of the sequence, but does not interpret the match/mismatch operators. It seems like the SAM1.4 CIGAR operations don't completely negate the MD tag, only when you don't care about the exact sequence of the reference.

Anyway, I cut a new release (0.1.2) which should be on PyPI in a few minutes.

**Comment by bede** [comment]
Sorry not to be clearer – yes it only means that the MD is redundant for a few specific applications. Still needed much of the time.

**Comment by mdshw5** [comment]
You got me excited, since it really bothers me that we have two strings (CIGAR and MD) that basically represent the same information in a slightly different way.

**Comment by bede** [comment]
Ah yes. At least we can now count indels *and* substitutions without the MD this way.

**Comment by bede** [comment]
This was addressed in 2020
